argilla/src/argilla/records/_io/_datasets.py

```diff
@@ -58,11 +60,65 @@ def _record_dicts_from_datasets(dataset: HFDataset) -> List[Dict[str, Union[str,
         Returns:
             Generator[Dict[str, Union[str, float, int, list]], None, None]: A generator of dictionaries to be passed to DatasetRecords.add or DatasetRecords.update.
         """
-        record_dicts = []
+        media_features = HFDatasetsIO._get_media_features(dataset)
+        if media_features:
+            dataset = HFDatasetsIO._cast_images_as_urls(hf_dataset=dataset, columns=media_features)
         try:
             dataset: IterableDataset = dataset.to_iterable_dataset()
         except AttributeError:
             pass
+        record_dicts = []
         for example in dataset:
             record_dicts.append(example)
         return record_dicts
+
+    @staticmethod
+    def _get_media_features(dataset: "HFDataset") -> List[str]:
+        """Check if the Hugging Face dataset contains image features.
+
+        Parameters:
+            hf_dataset (HFDataset): The Hugging Face dataset to check.
+
+        Returns:
+            bool: True if the Hugging Face dataset contains image features, False otherwise.
+        """
+        media_features = []
+        for name, feature in dataset.features.items():
+            if isinstance(feature, Image):
+                media_features.append(name)
+        return media_features
```

Comment:
list-comprehension approach. Feel free to apply this change.
```suggestion
        media_features = [
           name
           for name, feature in dataset.features.items()
           isinstance(feature, Image)
        ]
        return media_features
```